src/Stratis.Bitcoin/Utilities/Guard.cs

```diff
@@ -63,5 +82,38 @@ public static string NotEmpty(string value, string parameterName)
 
             return value;
         }
+
+        /// <summary>
+        /// Checks a <see cref="string"/> is not null or empty.
+        /// </summary>
+        /// <param name="value">The string to check.</param>
+        public static void NotEmpty(string value)
+        {
+            if (ReferenceEquals(value, null))
+            {
+                string name = GetOriginalVariableName(value);
+                throw new ArgumentNullException(name);
+            }
+
+            if (value.Trim().Length == 0)
+            {
+                string name = GetOriginalVariableName(value);
+                throw new ArgumentException($"The string parameter {name} cannot be empty.");
+            }
+        }
+
+        /// <summary>
+        /// Gets the name of the original variable.
+        /// </summary>
+        /// <param name="obj">The variable to get the name of.</param>
+        private static string GetOriginalVariableName(object obj)
+        {
+            StackFrame stackFrame = new StackTrace(true).GetFrame(2);
+            var file = new StreamReader(stackFrame.GetFileName());
+            for (int i = 0; i < stackFrame.GetFileLineNumber() - 1; i++)
+                file.ReadLine();
+            string varName = file.ReadLine().Split(new char[] { '(', ')' })[1];
+            return varName;
```

Comment:
are we allowed to use var in the name here ? 😺

Reply:
Lol.
But changed to `name`